src/AutoRest.CSharp/Common/Generation/Writers/ClientWriter.cs

```diff
@@ -116,5 +119,37 @@ protected void WriteDiagnosticScope(CodeWriter writer, Diagnostic diagnostic, st
                 writer.Line($"throw;");
             }
         }
+
+        protected Parameter[] GetNonPathParameters(RestClientMethod clientMethod)
```

Comment:
yes, but the functionality is applicable in both if anyone wants this.  If we don't do this we need to create a new base just to hold this one method since the various classes in mgmt all inherit from clientwriter.cs

Reply:
I think this is useful helper methods that does NOT look out of place here.